dataplane/linux/int_dataplane.go

```diff
@@ -341,23 +338,17 @@ func NewIntDataplaneDriver(config Config) *InternalDataplane {
 	dp.iptablesFilterTables = append(dp.iptablesFilterTables, filterTableV4)
 	dp.ipSets = append(dp.ipSets, ipSetsV4)
 
-	routeTableV4 := routetable.New(config.RulesConfig.WorkloadIfacePrefixes, 4, false, config.NetlinkTimeout, config.DeviceRouteSourceAddress, config.DeviceRouteProtocol, config.RemoveExternalRoutes)
-	dp.routeTables = append(dp.routeTables, routeTableV4)
-
 	if config.RulesConfig.VXLANEnabled {
-		routeTableVXLAN := routetable.New([]string{"vxlan.calico"}, 4, true, config.NetlinkTimeout, config.DeviceRouteSourceAddress, config.DeviceRouteProtocol, config.RemoveExternalRoutes)
-		dp.routeTables = append(dp.routeTables, routeTableVXLAN)
+		routeTableVXLAN := routetable.New([]string{"vxlan.calico"}, 4, true, config.NetlinkTimeout, config.DeviceRouteSourceAddress,
+			config.DeviceRouteProtocol, true)
+
 		vxlanManager := newVXLANManager(
 			ipSetsV4,
-			config.MaxIPSetSize,
-			config.Hostname,
 			routeTableVXLAN,
 			"vxlan.calico",
-			config.RulesConfig.VXLANVNI,
-			config.RulesConfig.VXLANPort,
-			config.ExternalNodesCidrs,
+			config,
```

Comment:
Was switching to just pass the config a cleanup? Or is there another reason?

It's nice to be explicit in the arguments here rather than passing the entire config struct. While it _does_ mean we have a longer constructor, it also means we know exactly what the VXLAN manager requires in order to operate. It's a fairly minor point though - WDYT?

Reply:
Yeah, I see that you pass it down pretty far. I guess this is probably the right change to make.

Reply:
Since the parent route table is now created in the vxlan manager we'd needed more information from the config, so I just added the config instead of adding more parameters, then realised there was no point in passing both the config and the explicit parameters from the config, so I just removed the other ones.

I think it's OK sometimes to have longer constructors, but as a preference I don't like them, I found it's more difficult to code with them, and I prefer hiding bigger constructors behind builder  patterns or allow config parameters. This said, I will defer to whatever has been common / agreed upon in the core team as this is more coding style specific.

Reply:
> but as a preference I don't like them, I found it's more difficult to code with them, and I prefer hiding bigger constructors behind builder patterns or allow config parameters

Yep, I agree completely. My though was more around passing a more targeted subset of the config. e.g, by defining a new `VXLANManagerConfig` struct or similar. 

But, I'm happy with the current approach after reading a bit further on.